awswrangler/mysql.py

```diff
@@ -323,6 +334,16 @@ def to_sql(
     """
     if df.empty is True:
         raise exceptions.EmptyDataFrame()
+    if mode.strip().lower() not in [
+        "append",
+        "overwrite",
+        "upsert_replace_into",
+        "upsert_duplicate_key",
+        "upsert_distinct",
+    ]:
+        raise exceptions.InvalidArgumentValue(
+            "mode must be one of append, overwrite, upsert_replace_into, upsert_duplicate_key, upsert_distinct"
+        )
```

Comment:
How about we simplify this to:
```
    mode = mode.strip().lower()
    modes = [
        "append",
        "overwrite",
        "upsert_replace_into",
        "upsert_duplicate_key",
        "upsert_distinct",
    ]
    if mode not in modes:
        raise exceptions.InvalidArgumentValue(
            f"mode must be one of {', '.join(modes)}"
        )
```
This way we only defines the list of modes once and likewise only have to perform the `strip().lower()` operation once on mode